sdk/ai/azure-ai-inference/src/main/java/com/azure/ai/inference/models/ChatCompletionsOptions.java

```diff
@@ -0,0 +1,604 @@
+     */
+    @Generated
+    private String model;
+
+    /*
+     * Controls what happens if extra parameters, undefined by the REST API,
+     * are passed in the JSON request payload.
+     * This sets the HTTP request header `extra-parameters`.
+     */
+    @Generated
+    private ExtraParameters extraParams;
+
+    /**
+     * Creates an instance of ChatCompletionsOptions class.
+     *
+     * @param messages the messages value to set.
+     */
+    @Generated
+    public ChatCompletionsOptions(List<ChatRequestMessage> messages) {
+        this.messages = messages;
+    }
+
+    /**
+     * Get the messages property: The collection of context messages associated with this chat completions request.
+     * Typical usage begins with a chat message for the System role that provides instructions for
+     * the behavior of the assistant, followed by alternating messages between the User and
+     * Assistant roles.
+     *
+     * @return the messages value.
+     */
+    @Generated
+    public List<ChatRequestMessage> getMessages() {
+        return this.messages;
+    }
+
+    /**
+     * Get the frequencyPenalty property: A value that influences the probability of generated tokens appearing based on
+     * their cumulative
+     * frequency in generated text.
+     * Positive values will make tokens less likely to appear as their frequency increases and
+     * decrease the likelihood of the model repeating the same statements verbatim.
+     * Supported range is [-2, 2].
+     *
+     * @return the frequencyPenalty value.
+     */
+    @Generated
+    public Double getFrequencyPenalty() {
+        return this.frequencyPenalty;
+    }
+
+    /**
+     * Set the frequencyPenalty property: A value that influences the probability of generated tokens appearing based on
+     * their cumulative
+     * frequency in generated text.
+     * Positive values will make tokens less likely to appear as their frequency increases and
+     * decrease the likelihood of the model repeating the same statements verbatim.
+     * Supported range is [-2, 2].
+     *
+     * @param frequencyPenalty the frequencyPenalty value to set.
+     * @return the ChatCompletionsOptions object itself.
+     */
+    @Generated
+    public ChatCompletionsOptions setFrequencyPenalty(Double frequencyPenalty) {
+        this.frequencyPenalty = frequencyPenalty;
+        return this;
+    }
+
+    /**
+     * Get the stream property: A value indicating whether chat completions should be streamed for this request.
+     *
+     * @return the stream value.
+     */
+    @Generated
+    public Boolean isStream() {
+        return this.stream;
+    }
+
+    /**
+     * Set the stream property: A value indicating whether chat completions should be streamed for this request.
+     *
+     * @param stream the stream value to set.
+     * @return the ChatCompletionsOptions object itself.
+     */
+    @Generated
+    public ChatCompletionsOptions setStream(Boolean stream) {
```

Comment:
This should not be public as we have separate methods for streaming and non-streaming responses in the client.

Reply:
How can this be refactored? The completeStream methods have to set the stream parameter in the ChatCompletionsOptions class somehow, since stream corresponds to a field in the post body request.

Reply:
[Added access helper](https://github.com/Azure/azure-sdk-for-java/pull/41482/commits/7defa2a6bb0fa24f4a9e2a25dab45545fd04e8c1)